pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/ServerCnx.java

```diff
@@ -301,9 +270,19 @@ protected void handleConsumerStats(CommandConsumerStats commandConsumerStats) {
         commandConsumerStatsResponseBuilder.setAddress(consumerStats.address);
         commandConsumerStatsResponseBuilder.setConnectedSince(consumerStats.connectedSince);
         
-        commandConsumerStatsResponseBuilder.setType(subscription.getTypeString());
-        commandConsumerStatsResponseBuilder.setMsgRateExpired(subscription.getExpiredMessageRate());
+        Subscription subscription = consumer.getSubscription();
         commandConsumerStatsResponseBuilder.setMsgBacklog(subscription.getNumberOfEntriesInBacklog());
+        
+        String subsTypeString = "Null";
+        double msgExpireRate = -1; 
+        if (subscription instanceof PersistentSubscription) {
+            msgExpireRate = ((PersistentSubscription)subscription).getExpiredMessageRate();
```

Comment:
instead casting to `PersistentSubscription` can't we just add these methods into `Subscription` ?